### PR TITLE
#162199433 Create Featured Article Section

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
Create Featured Article Section

#### Description of Task to be completed?
- Create the featured article section component for the landing page
- Implement redux-saga in consuming the API route
- Integrate the Navbar component.

#### How should this be manually tested?
On command line i;
- git clone this repo
- Run `npm start` from the `haven-ah-frontend` directory
- Visit http://localhost:5001

#### What are the relevant pivotal tracker stories?
[#162199433](https://www.pivotaltracker.com/story/show/162199433)

#### Screenshots (if appropriate)>
ezgif com-optimize

### CODE REVIEW

File: src/sagas/heroArticleSaga.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { takeLatest, call, put } from 'redux-saga/effects';
+import { HERO_ARTICLE_REQUEST } from '../actionTypes/heroActionType';
+import {
+  heroArticleFailureAction,
+  heroArticleSuccessAction,
+} from '../actions/heroArticleAction';
+
+const fetchHeroArticle = () => {
+  return axios.get(
+    'https://haven-ah-backend-staging.herokuapp.com/api/v1/articles/featured'
```
Comment (seyi-adeleke) [inline src/sagas/heroArticleSaga.js]: We should not have hardcoded urls.

Comment (TheoOkafor) [reply src/sagas/heroArticleSaga.js]: Okay

File: src/components/containers/navbar/navbar.jsx [hunk 2]
```diff
@@ -1,43 +1,76 @@
...
-          </Link>
-          <div
-            className="navbar-burger burger"
-            data-target="ahNavbar"
-            ref={burgerRef}
-          >
-            <span />
-            <span />
-            <span />
+class NavBar extends PureComponent {
```
Comment (seyi-adeleke) [inline src/components/containers/navbar/navbar.jsx]: lets not be overzealous in our use of PureComponent. Can lead to unexpected bugs

File: src/components/containers/navbar/navbar.jsx [hunk 3]
```diff
@@ -1,43 +1,76 @@
...
+  constructor(props) {
+    super(props);
+    this.scrollHandler = this.scrollHandler.bind(this);
```
Comment (seyi-adeleke) [inline src/components/containers/navbar/navbar.jsx]: if scrollHandler is an arrow function you wont have need for this constructor

File: src/components/containers/navbar/navbar.jsx [hunk 4]
```diff
@@ -1,43 +1,76 @@
...
+  }
+
+  toggleNavbar = () => {
+    this.refs.navMenu.classList.toggle('is-active');
+  }
+
+  scrollHandler(event) {
+    const { navbar } = this.refs;
+    const { pageYOffset } = event.currentTarget;
+    if (pageYOffset >= 100) {
```
Comment (seyi-adeleke) [inline src/components/containers/navbar/navbar.jsx:27]: Looks familiar 🤔  🤣

File: src/components/hero-section/HeroSection.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import React, { PureComponent } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import 'bulma/css/bulma.min.css';
+import './HeroSection.scss';
+import { heroArticleAction } from '../../actions/heroArticleAction';
+
+class HeroSection extends PureComponent {
```
Comment (seyi-adeleke) [inline src/components/hero-section/HeroSection.jsx]: same comment about PureComponent

File: src/components/views/HomePage.jsx [hunk 6]
```diff
@@ -1,34 +1,14 @@
...
-
-  toggleNavbar = () => {
-    this.navElement.classList.toggle('is-active');
-  }
-
   render() {
     return (
-      <Navbar burgerRef={this.getBurger} navRef={this.getNav} isLoggedIn />
+      <header className="banner">
+        <Navbar navStyle="transparent" isLoggedIn/>
```
Comment (seyi-adeleke) [inline src/components/views/HomePage.jsx:9]: nice

File: src/store.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import rootReducer from './reducers';
+import rootSaga from './sagas';
+
+const sagaMiddleware = createSagaMiddleware();
+const initialState = {};
+
+const store = createStore(
+  rootReducer,
+  initialState,
+  composeWithDevTools(
```
Comment (seyi-adeleke) [inline src/store.js:13]: can we possibly only use this in a dev environment?

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: src/store.js (hunk 7 above)
Comment (TheoOkafor) [reply src/store.js:13]: Yes

File: src/store.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { createStore, applyMiddleware } from 'redux';
+import createSagaMiddleware from 'redux-saga';
+import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
```
Comment (TheoOkafor) [inline src/store.js:3]: @seyi-adeleke we are using the `developmentOnly` version

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**